Kucoin connector implemented.

ItIsOHM opened the pull request:
I have implemented the public connector with many good practices like types and guards and all other things. Working on the private connector now.

@skltrading please look into this and let me know of any issues :)

ItIsOHM committed:
implemented public connector with most requirements satisfied.

skltrading commented:
Hey Rhythm,

Thank you for your submission, please book a slot in my calendar
https://calendar.google.com/calendar/u/0/appointments/schedules/AcZssZ0xXATdE83dGytAzsIYVTyl-ZhU4xYymigJDElyGlypS4YPYOYlGNgLLwDqyD2uNF-AEZjcF0Hv


Kind regards
Sam

On Wed, 16 Oct 2024 at 18:15, Rhythm Garg ***@***.***> wrote:

> I have implemented the public connector with many good practices like
> types and guards and all other things. Working on the private connector now.
>
> @skltrading <https://github.com/skltrading> please look into this and let
> me know of any issues :)
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/skltrading/connectorTrial/pull/15
> Commit Summary
>
>    - e8ddd59
>    <https://github.com/skltrading/connectorTrial/pull/15/commits/e8ddd59cd41deffe49bbecd0bbcc0bb03f9140bc>
>    implemented public connector with most requirements satisfied.
>
> File Changes
>
> (18 files <https://github.com/skltrading/connectorTrial/pull/15/files>)
>
>    - *M* .gitignore
>    <https://github.com/skltrading/connectorTrial/pull/15/files#diff-bc37d034bad564583790a46f19d807abfe519c5671395fd494d8cce506c42947>
>    (6)
>    - *A* KuCoin (Rhythm)/package-lock.json
>    <https://github.com/skltrading/connectorTrial/pull/15/files#diff-ca315bb83f140c990af0f1320f2672c17f29e1851b8320dbad6966e6f787ecef>
>    (734)
>    - *A* KuCoin (Rhythm)/package.json
>    <https://github.com/skltrading/connectorTrial/pull/15/files#diff-0d4857d9da31bd2551f45bf8a8bf112102ed74451a6683129431af3baf529aa5>
>    (17)
>    - *D* KuCoin
>    (Rhythm)/src/connectors/public/kucoin-spot-public-connector.ts
>    <https://github.com/skltrading/connectorTrial/pull/15/files#diff-2320f31c522df9d1cb9441cfd93da8bacf0fe0172028d5831b8eea96c7951563>
>    (0)
>    - *D* KuCoin (Rhythm)/src/connectors/types.ts
>    <https://github.com/skltrading/connectorTrial/pull/15/files#diff-83d6e2bdf510a03dd4a8fa43db4a1ced156427125c1c6f30fa893fd2b4786b66>
>    (0)
>    - *A* KuCoin (Rhythm)/src/index.ts
>    <https://github.com/skltrading/connectorTrial/pull/15/files#diff-cd58727b1fe0b82b2beb90cd8a112ec792fe33e362219e12e3d124a362c067d9>
>    (35)
>    - *R* KuCoin (Rhythm)/src/kucoin-spot.ts
>    <https://github.com/skltrading/connectorTrial/pull/15/files#diff-2db96bc54d7b28ae947ad2da571ca58566f2e2d856ba3e748459f9c0b5988cd1>
>    (0)
>    - *A* KuCoin (Rhythm)/src/logger.ts
>    <https://github.com/skltrading/connectorTrial/pull/15/files#diff-caf84fd6ccf51a50ef269c952467f3e50ffce3101793178f34e53fe69a7d4903>
>    (19)
>    - *A* KuCoin (Rhythm)/src/order-book-manager.ts
>    <https://github.com/skltrading/connectorTrial/pull/15/files#diff-d07cf01e6944a9ad73db259dd040193cc98d47d1a735d4dc988ed502c1e17a62>
>    (45)
>    - *R* KuCoin (Rhythm)/src/private/kucoin-spot-private-connector.ts
>    <https://github.com/skltrading/connectorTrial/pull/15/files#diff-79eac24ca6e60e8f997fb5a834592a63d595aacf027f608b03e9a91fd33f9987>
>    (0)
>    - *A* KuCoin (Rhythm)/src/processors.ts
>    <https://github.com/skltrading/connectorTrial/pull/15/files#diff-6c49c25c4494dc7923b0bded054b25209dfde0f7feb1021b00523eb894fd6030>
>    (47)
>    - *A* KuCoin (Rhythm)/src/public/event-handlers.ts
>    <https://github.com/skltrading/connectorTrial/pull/15/files#diff-f2896b23e95da00611181cde901997172eea4d7d1151a59662caae329ada8cb4>
>    (17)
>    - *A* KuCoin (Rhythm)/src/public/kucoin-spot-public-connector.ts
>    <https://github.com/skltrading/connectorTrial/pull/15/files#diff-d49f0b1cb8bd052a31e8d4e0d46eebb317953c65c9a4c037242e36ba8068a573>
>    (244)
>    - *A* KuCoin (Rhythm)/src/typeguards.ts
>    <https://github.com/skltrading/connectorTrial/pull/15/files#diff-92ed3c26e4a1bda20fa5c79e24d6802f79fe9b518b4d65e8eb6bf1b570a19069>
>    (25)
>    - *A* KuCoin (Rhythm)/src/types.ts
>    <https://github.com/skltrading/connectorTrial/pull/15/files#diff-4700d2aed1a1955e8a659948e430312e6be4d4770dc3d0a2784cc021860ac8c2>
>    (51)
>    - *A* KuCoin (Rhythm)/tsconfig.json
>    <https://github.com/skltrading/connectorTrial/pull/15/files#diff-f3a1dec9dc1daab9ac06154836277b20a37632bb0787be66bd375af5311efb96>
>    (11)
>    - *D* package-lock.json
>    <https://github.com/skltrading/connectorTrial/pull/15/files#diff-053150b640a7ce75eff69d1a22cae7f0f94ad64ce9a855db544dda0929316519>
>    (32)
>    - *D* package.json
>    <https://github.com/skltrading/connectorTrial/pull/15/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519>
>    (5)
>
> Patch Links:
>
>    - https://github.com/skltrading/connectorTrial/pull/15.patch
>    - https://github.com/skltrading/connectorTrial/pull/15.diff
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/skltrading/connectorTrial/pull/15>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BADRNUW43BAKWVIXDWTUMGDZ32NJZAVCNFSM6AAAAABQB6CT6WVHI2DSMVQWIX3LMV43ASLTON2WKOZSGU4TENJTGI3DQOI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
[image: sc]
[image: icon]
*Samuel Lannaghan*
*Director*
[image: icon] +447469877536
[image: icon] ***@***.***
[image: icon] www.sklventures.io <https://sklventures.io/>
Institutional Infrastructure for Digital Assets
[image: sc] <https://www.sklventures.io/>

[image: sc] <https://calendar.app.google/8f9ArtSWgoc8piiQ6>

[image: sc] <https://sklventures.io/>

[image: sc] <https://www.linkedin.com/in/sam-lannaghan-152355231/>  [image:
sc] <https://twitter.com/SKL_MM>
The content of this email is confidential and intended for the recipient
specified in message only. It is strictly forbidden to share any part of
this message with any third party, without a written consent of the sender.
If you received this message by mistake, please reply to this message and
follow with its deletion, so that we can ensure such a mistake does not
occur in the future.

ItIsOHM committed:
Implemented Private connector.

ItIsOHM committed:
minor touch-ups

ItIsOHM committed:
Cleaning up and formatting. Tried implementing Zod but realised it's not really needed so back to using typeguards.

ItIsOHM commented:
Hey @skltrading, please take a look at this PR and merge if it looks good to you :)

ItIsOHM committed:
Implement rate limiter class and update KuCoin connectors

- Implemented the RateLimiter class to limit the number of requests made to the Kucoin API per interval.
- Updated the KuCoinPublicConnector and KuCoinPrivateConnector classes to add and remove subscription channels.
- Added examples for using the public and private connectors in the index.ts file.
- Created a test-index.ts file to test rate limiting for REST API requests in the KuCoinPrivateConnector class.

ItIsOHM committed:
Fixed a bug where level 3 matches were not logging correctly.
